stdlib/LinearAlgebra/test/bidiag.jl

```diff
@@ -468,4 +468,45 @@ end
     end
 end
 
+@testset "multiplication of bidiagonal and triangular matrix" begin
+    n = 5
+    for eltyB in (Int, ComplexF64)
+        if eltyB == Int
+            BU = Bidiagonal(rand(1:7, n), rand(1:7, n - 1), :U)
+            BL = Bidiagonal(rand(1:7, n), rand(1:7, n - 1), :L)
+        else
+            BU = Bidiagonal(randn(eltyB, n), randn(eltyB, n - 1), :U)
+            BL = Bidiagonal(randn(eltyB, n), randn(eltyB, n - 1), :L)
+        end
+        for eltyT in (Int, ComplexF64)
+            for TriT in (LowerTriangular, UnitLowerTriangular, UpperTriangular, UnitUpperTriangular)
+                if eltyT == Int
+                    T = TriT(rand(1:7, n, n))
+                else
+                    T = TriT(randn(eltyT, n, n))
+                end
+                for B in (BU, BL)
+                    MB = Matrix(B)
+                    MT = Matrix(T)
+                    @test B * T ≈ MB * MT
+                    @test B * T' ≈ MB * MT'
+                    @test B * transpose(T) ≈ MB * transpose(MT)
+                    @test B' * T ≈ MB' * MT
+                    @test transpose(B) * T ≈ transpose(MB) * MT
+                    @test B' * T' ≈ MB' * MT'
+                    @test transpose(B) * transpose(T) ≈ transpose(MB) * transpose(MT)
+
+                    @test T * B ≈ MT * MB
+                    @test T * B' ≈ MT * MB'
+                    @test T * transpose(B) ≈ MT * transpose(MB)
+                    @test T' * B ≈ MT' * MB
+                    @test transpose(T) * B ≈ transpose(MT) * MB
+                    @test T' * B' ≈ MT' * MB'
+                    @test transpose(T) * transpose(B) ≈ transpose(MT) * transpose(MB)
```

Comment:
As in #33579, I'd suggest to replace this with a double loop over two transforms.